contracts/token/ERC721/ERC721.sol

```diff
@@ -331,8 +331,24 @@ contract ERC721 is Context, ERC165, IERC721 {
             return true;
         }
 
-        bytes4 retval = IERC721Receiver(to).onERC721Received(_msgSender(), from, tokenId, _data);
-        return (retval == _ERC721_RECEIVED);
+        bytes memory payload = abi.encodeWithSignature(
+            "onERC721Received(address,address,uint256,bytes)",
+            msg.sender,
```

Comment:
Please use `_msgSender()` instead, for compatibility with `Context`. This is used for integration with the [GSN](https://gasstation.network).

```suggestion
            _msgSender(),
```